docs/developer_workflow.md

```diff
@@ -8,36 +8,26 @@ cd cluster-api-provider-nutanix
 
 ## Build source code
 <pre>
-make docker-build
-</pre>
-
-## Create test management cluster
-<pre>
-make kind-create
-</pre>
-
-## Prepare local clusterctl
-<pre>
-make prepare-local-clusterctl
+make build-snapshot
```

Comment:
I don't think we even need the build anymore, the `dev.run-on-kind` target already runs `release-snapshot`